src/Engine/Audio/SteamAudio.cpp

```diff
@@ -1,59 +1,87 @@
+    directBuffer.deinterleavedBuffer = NULL;
 }
 
 SteamAudio::~SteamAudio() {
-    if (context != nullptr)
-        delete context;
-}
+    Clear();
 
-void SteamAudio::Process(IPLAudioBuffer input, IPLVector3 * playerPos, IPLVector3 * playerDir, IPLVector3 * playerUp, IPLVector3 * sourcePos, float sourceRadius) {
-    // Direct Processing
+    delete[] indirectBuffer.interleavedBuffer;
+    delete[] directBuffer.interleavedBuffer;
+}
 
-    // Indirect Processing
+void SteamAudio::Process(std::vector<SteamAudio::SoundSourceInfo>& inputs, IPLAudioBuffer& output) {
 
-    //MixAudio(direct, indirect)
+    std::vector<IPLAudioBuffer> audioBuffers;
+    audioBuffers.resize(inputs.size());
+    for (std::size_t i = 0; i < inputs.size(); ++i) {
+        SteamAudio::SoundSourceInfo& input = inputs[i];
+        audioBuffers[i] = input.renderers->Process(input.buffer, playerPos, playerDir, playerUp, input.position, input.radius);
+    }
 
-    processedBuffers.push_back(input);
+    assert(!audioBuffers.empty());
+    output.format = audioBuffers[0].format;
+    iplMixAudioBuffers(audioBuffers.size(), audioBuffers.data(), directBuffer);
+    iplGetMixedEnvironmentalAudio(envRenderer, playerPos, playerDir, playerUp, indirectBuffer);
+    std::vector<IPLAudioBuffer> mixBuffers{ directBuffer, indirectBuffer };
+    iplMixAudioBuffers(mixBuffers.size(), mixBuffers.data(), output);
+}
 
+void SteamAudio::SetPlayer(IPLVector3 pos, IPLVector3 dir, IPLVector3 up) {
+    playerPos = pos;
+    playerDir = dir;
+    playerUp = up;
 }
 
-void SteamAudio::GetFinalMix(IPLAudioBuffer* finalBuf, uint32_t* numSamples) {
-    // No audio is playing, return emptiness, let SoundManager figure it out.
-    if (processedBuffers.size() == 0) {
-        *numSamples = 0;
-        finalBuf = nullptr;
-        return;
-    }
+void SteamAudio::CreateRenderers(IPLhandle environment) {
+    Clear();
 
-    // Move the buffers into an array (needed for steam audios mixing)
-    IPLAudioBuffer* buffers = new IPLAudioBuffer[processedBuffers.size()];
-    for (std::size_t i = 0; i < processedBuffers.size(); i++) {
-        buffers[i] = processedBuffers[i];
-    }
+    this->environment = environment;
 
-    finalBuf->format = buffers[0].format;
-    finalBuf->interleavedBuffer = new float[buffers[0].numSamples];
-    finalBuf->numSamples = buffers[0].numSamples;
-    iplMixAudioBuffers(processedBuffers.size(), buffers, *finalBuf);
-    *numSamples = finalBuf->numSamples;
+    IPLRenderingSettings settings;
+    settings.frameSize = CHUNK_SIZE;
+    settings.samplingRate = SAMPLE_RATE;
+    settings.convolutionType = IPL_CONVOLUTIONTYPE_PHONON;
 
-    // Clean up
-    for (IPLAudioBuffer buf : processedBuffers) {
-        if (buf.interleavedBuffer != nullptr)
-            delete[] buf.interleavedBuffer;
+    IPLerror err = iplCreateEnvironmentalRenderer(IPLContext{ nullptr,nullptr,nullptr }, environment, settings, outputFormat, NULL, NULL, &envRenderer);
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') comma should be followed by whitespace [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=vera%2B%2B%3AT009)